### PR TITLE
fix: delegation id randomly not found

### LINKED ISSUE #1424: [Task]: Delegation ID not found
### Task description

cant find any delegation ID, eg `0xcd4fe26ec9d9ef86320aef402da554cc4717fb97e5ddae73e57b000d3e4defaa`  cant find this but the delegation is ID in the output
image

> All my other Delegation Outputs show up fine in the Explorer. All the ones that work have Start Epoch 4, while the one that doesn't work has start epoch 3 - not sure if that's somehow the cause though.

> When I try to search for my Delegation Output from Firefly it says "Nft output not found" (repeating this action makes it show "Anchor output not found").

### PR DESCRIPTION
**Pull request by panteleymonchuk**
# Description of change

Closes #1424

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/src/utils/nova/searchExecutor.ts [hunk 1]
```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
...
+                    this.apiService
+                        .delegationDetails(searchQuery.delegationId)
+                        .then((response) => {
+                            if (response.output) {
+                                promisesResult = {
+                                    output: response.output,
+                                };
+                                resolve();
+                            } else {
+                                reject(new Error("Anchor id fetch failed"));
```
Comment (msarcev) [inline api/src/utils/nova/searchExecutor.ts:146]: "Delegation Id fetch failed" here and four lines below

File: api/src/utils/nova/searchExecutor.ts [hunk 2]
```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
...
-                            slotIndex: String(searchQuery.slotIndex),
-                        };
-                    },
-                    "Slot commitment fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .getSlotCommitment(searchQuery.slotIndex)
+                        .then((response) => {
+                            if (searchQuery.slotIndex) {
```
Comment (msarcev) [inline api/src/utils/nova/searchExecutor.ts:206]: Wrong, here is `response.something` not `searchQuery.slotIndex`

File: api/src/utils/nova/searchExecutor.ts [hunk 3]
```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
...
-                    "Anchor id fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .anchorDetails(searchQuery.anchorId)
+                        .then((response) => {
+                            if (response.anchorOutputDetails) {
+                                promisesResult = {
+                                    anchorId: searchQuery.anchorId,
+                                };
```
Comment (msarcev) [inline api/src/utils/nova/searchExecutor.ts:122]: `resolve()` missing.

File: api/src/utils/nova/searchExecutor.ts [hunk 4]
```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
...
-                            slotIndex: String(result.slot.slot),
-                        };
-                    },
-                    "Slot commitment fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .getCommitment(searchQuery.slotCommitmentId)
+                        .then((response) => {
+                            if (response?.slot?.slot) {
```
Comment (msarcev) [inline api/src/utils/nova/searchExecutor.ts:228]: Here i'm concerned with the case where `response.slot.slot === 0`, which could be a valid slot, but the condition will no enter (falsy). Maybe we can just be content with `if (response?.slot)` to be enough.

File: api/src/utils/nova/searchExecutor.ts [hunk 5]
```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
...
-                            };
-                        }
-                    },
-                    "Tagged details fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .taggedOutputs(searchQuery.tag)
+                        .then((response) => {
+                            if (!response.basicOutputs.error || !response.nftOutputs.error) {
```
Comment (msarcev) [inline api/src/utils/nova/searchExecutor.ts:250]: I don't trust out "errors" too much.
Consider just accepting `if (response?.basicOutputs?.outputs || response?.nftOutputs?.outputs)` to be a strong enough condition to resolve the promise.

**Review by msarcev — CHANGES_REQUESTED** [review]

**Review by msarcev — APPROVED** [review]

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**